File: charles-joma/wk01/5-fri/MTA/js/mta2.js
const lines = {
    lineL: ['Times Square' , ' 34th ', ' 28th ', ' 23rd ', ' Union Square ', ' 8th '],
    lineN: [' Grand central ', ' 33rd ', '28th ', ' 23rd ', ' Union Square ',' Astor place '],
    line6: [' 8th ', ' 6th ', ' Union Square ', ' 3rd ', ' 1st '],
}

const loopThroughStops = function(startLine, startStop, endLine, endStop){
    let numberOfStops = 0;

    let journey = [];

    let lineSelect = lines[startLine];

    let startSelect = lines[startLine].indexOf(startStop)

    let endLineSelect  = lines[endLine];

    let endSelect = lines[endLine].indexOf(endStop);

if(startLine === lines.lineL || lineSelect === lines.lineN || lineSelect === lines.line6)
    for(let i = startSelect; i <= lineSelect.length; i++){
        journey.push(lineSelect[i])
        numberOfStops++
        break
    } 
    if(startStop != endStop){
        for(let i = endLineSelect.length-1; i >= endSelect; i--){
            journey.push(endLineSelect[i])
            numberOfStops++

        }
    }
    return `You Must Travel through the Following stops on ${startLine}: ${journey}in order to get to${endStop} \nUnion Square being your change over \nMaking the journey a total of ${numberOfStops} stops`
}

console.log(loopThroughStops("line6"," 3rd ","lineL"," 34th "))

/*
console.log(lineSelect)
    console.log(startSelect)
    console.log(endLineSelect)
    console.log(endSelect)
    */

File: edward-do/wk04/2-tues/mta2.rb
def plan_trip # start_line, start_stop, dest_line, dest_stop
    print "Welcome to the MTA. Please enter your starting line: "
    start_line = gets.chomp 
    print "Please enter your starting stop: "
    start_stop = gets.chomp 
    print "Please enter your destination line: "
    dest_line = gets.chomp 
    print "Please enter your destination stop: "
    dest_stop = gets.chomp

    lines = {
    "L" => ['1st', '3rd', 'Union Square', '6th', '8th'],
    "6" => ['Astor Place', 'Union Square', '23rd', '28th', '33rd', 'Grand Central'],
    "N" => ['8th', 'Union Square', '23rd', '28th', '34th', 'Times Square']
    }

    first_message = "You must travel through the following stops on the #{start_line} line: "
    second_message = "Your journey continues through the following stops: "
    change_at_us = "Change at Union Square."

    start_line_array = lines[start_line]
    dest_line_array = lines[dest_line]
    start_line_stop = start_line_array.index start_stop
    dest_line_stop = dest_line_array.index dest_stop


    puts first_message + start_line[start_line_stop .. dest_line_stop]
    p dest_line_array.index dest_stop
end

plan_trip